src/treelearner/gpu_tree_learner.cpp

```diff
@@ -1,15 +1,14 @@
 #ifdef USE_GPU
 #include "gpu_tree_learner.h"
-#include "../io/dense_bin.hpp"
-#include "../io/dense_nbits_bin.hpp"
 
-#include <LightGBM/utils/array_args.h>
-#include <LightGBM/network.h>
 #include <LightGBM/bin.h>
+#include <LightGBM/network.h>
+#include <LightGBM/utils/array_args.h>
 
 #include <algorithm>
-#include <vector>
 
+#include "../io/dense_bin.hpp"
+#include "../io/dense_nbits_bin.hpp"
```

Comment:
I don't know how to bring this into consistency with the guide:
> All of a project's header files should be listed as descendants of the project's source directory without use of UNIX directory shortcuts . (the current directory) or .. (the parent directory).

Reply:
I think it is okay to include the hpp file by this way.
ping @huanzhang12

Reply:
@guolinke Can this be merged without these fixes because this PR will produce many conflicts (too many files touched)? And later we'll create another PR for `..` in case @huanzhang12 will have good ideas.

Reply:
Sure